From 5484868d661acf8772ff867608225bd26d9b7ae4 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas@debian.org>
Date: Mon, 1 Nov 2021 10:08:08 +0100
Subject: [PATCH 04/11] scripts/mk/buildopts.mk: search once for parrallel= in
 DEB_BUILD_OPTIONS

Previously each expansion of DEB_BUILD_OPTION_PARALLEL was parsing
DEB_BUILD_OPTIONS.
---
 scripts/mk/buildopts.mk | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/scripts/mk/buildopts.mk b/scripts/mk/buildopts.mk
index 2aa65f5d0..50ffc5ba6 100644
--- a/scripts/mk/buildopts.mk
+++ b/scripts/mk/buildopts.mk
@@ -4,12 +4,16 @@
 # Defines the following variables:
 #
 #   DEB_BUILD_OPTION_PARALLEL: the argument for the parallel=N option.
+#     $(DEB_BUILD_OPTIONS)                "parallel=2"  "parallel="  ""
+#     $(DEB_BUILD_OPTION_PARALLEL)        "2"           ""           unset
+#     $(DEB_BUILD_OPTION_PARALLEL:%=-j%)  "-j2"         ""           ""
 
 ifndef /usr/share/dpkg/buildopts.mk_included
 /usr/share/dpkg/buildopts.mk_included = yes
 
-ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
-  DEB_BUILD_OPTION_PARALLEL = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+dpkg-buildopts_parallel := $(filter parallel=%,$(DEB_BUILD_OPTIONS))
+ifdef dpkg-buildopts_parallel
+  DEB_BUILD_OPTION_PARALLEL = $(patsubst parallel=%,%,$(dpkg-buildopts_parallel))
 endif
 
 endif # ifndef /usr/share/dpkg/buildopts.mk_included
-- 
2.39.2

